### PR TITLE
feat: check for key mutation when removing children

### PR DESCRIPTION
**Pull request by iisakkirotko**

### CODE REVIEW

**Comment by iisakkirotko** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/widgetti/reacton/35?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#35** <a href="https://app.graphite.dev/github/pr/widgetti/reacton/35?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#34** <a href="https://app.graphite.dev/github/pr/widgetti/reacton/34?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iisakkirotko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: reacton/core_test.py [hunk 1]
```diff
@@ -3186,3 +3186,25 @@ def Test():
     assert vbox.children[0].description == "1"
     assert vbox.children[1].description == "2"
     rc.close()
+
+
+def test_key_mutate_and_reuse(Container):
```
Comment (maartenbreddels) [inline reacton/core_test.py:3191]:
```suggestion
def test_key_mutate_protection(Container):
```

**Review by maartenbreddels — CHANGES_REQUESTED** [review]
Awesome to see this nice and clean PR after all this work :)
Please squash the 2 commits into 1, I see the test as part of the feature.
🥳

**Review by maartenbreddels — APPROVED** [review]

**Comment by iisakkirotko** [comment]
### Merge activity

* **Apr 26, 5:51 AM EDT**: @iisakkirotko started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/widgetti/reacton/35).

**Pull request merged**